kani-compiler/src/kani_middle/resolve.rs

```diff
@@ -445,3 +583,25 @@ fn resolve_in_type<'tcx>(
         })
         .ok_or_else(missing_item_err)
 }
+
+/// Resolves a function in a trait.
+fn resolve_in_trait<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    trait_id: DefId,
+    name: &str,
+) -> Result<DefId, ResolveError<'tcx>> {
+    debug!(?name, ?trait_id, "resolve_in_trait");
+    let missing_item_err =
+        || ResolveError::MissingItem { tcx, base: trait_id, unresolved: name.to_string() };
+    let trait_def = tcx.trait_def(trait_id);
+    // Try the inherent `impl` blocks (i.e., non-trait `impl`s).
+    tcx.associated_item_def_ids(trait_def.def_id)
+        .iter()
+        .copied()
+        .find(|item| {
+            let item_path = tcx.def_path_str(*item);
```

Comment:
Why not `tcx.item_name(item)` to avoid string operations?

Reply:
Because `def_path_str` returns a string suitable for user output.